backend/api/models.py

```diff
@@ -33,6 +36,28 @@ class FullMoodleSettings(BaseMoodleSettings):
     id: int
 
 
+class BaseCourseBuildSettings(BaseModel):
+    instructor_firstname: str
+    instructor_lastname: str
+    instructor_email: str
+    school_district: int
+
+
+class FullCourseBuildSettings(BaseCourseBuildSettings):
+    course_name: str
+    course_shortname: str
+    course_category: int
+    course_id: Optional[int] = None
+    course_enrollment_url: Optional[str] = None
+    course_enrollment_key: Optional[str] = None
+    academic_year: str
+    academic_year_short: str
+    base_course_id: int
+    creator: int
+    status: enum.Enum
```

Comment:
This should be a `str` type. In fact, we could even do:

```python
    status: Literal["CREATED", ""PROCESSING", "COMPLETED", "FAILED]
```

That way it'll validate for us too.

Reply:
If I make the type for `status` a `str`, then when I set the value of `status` as `CourseBuildStatus.CREATED` all my tests pass. However, If I make the type a `Literal` as shown above, I receive the following:

![Screenshot 2024-02-26 at 9 24 12 AM](https://github.com/openstax/rope/assets/37000585/5dc9bf9e-06e4-4f95-80e2-68c7ca51cf0d)

Reply:
I tested updating the schema as noted above and the tests pass. The revision checks that the value is a valid member of the string enum.

Reply:
Nice catch / find! :tada:  

Is the main difference the update saves a bit of manual serialization otherwise (e.g. saves us from having to do things like `status = str(CourseBuildStatus.CREATED)` ?

Reply:
Yes, that is one of the main perks of providing a subclass to the `enum`.